### DISCUSSION TITLE
View node as a sub-map

### DISCUSSION THREAD

**Discussion by ianikanov** [freeplane/freeplane#455] [Ideas]
Hi all,
A feature request / contribution proposal to be able to view a node as a sub-map. Not converting it into a separate map, but rather narrowing down your focus.

Let me give you a specific use-case where this would be useful, let's say you are working on multiple things in parallel (or worked historically) - for example a list of IT projects (I know a bit extreme, but still realistic). So you keep your map structure something like this:

The top map:

> Left side:
> - Inbound tasks
> - Knowledge management
> - Team management
> - Focus areas
> 
> Root item: My Projects
> 
> Right side:
> - Project 1
> - - Deadlines
> - - Business context
> - - Tasks
> - - Architecture
> - - People
> - Project 2
> - - <same structure>
> - ...
> - Archive
> - - Project X

The beauty of having this all in one map is to be able to cross-link the nodes and also apply common filter (for example to see all the tasks from all the areas) and many more, but the drawback is that when you want to narrow down the focus and work only on specific project you are only limited with one side expansion (further to the right) and have a lot of scrolling and visual confusion on where the current project ends and the next one begins.

The idea is to be able to work with this map as it is with all the cross links and filters, but on occasion choose a specific node as a new "virtual" root and focus only on that.

So for example above each project may be transformed into a virtual sub-map like this:

> Left side:
> - Deadlines
> - Tasks
> 
> Root item: Project 1
> 
> Right side:
> - Business context
> - Architecture
> - People

Happy to talk through idea in more details if needed. I can also contribute into developing of this feature, but would need somebody experienced in the Freeplane code to guide me.

**Comment by macmarrum** [comment]
Until your feature request is implemented, I would use `View->Hidden node` for the non-focused branches, applied by a script. This would leave room for using filters on the focused branch.

**Comment by EdoFro** [comment]
A time ago, there was a feature request that the free positioned nodes could behave as root nodes, so they could have child nodes to both sides.

Maybe both requests can be solved together. I see this new request similar to the previous one with just an added filter (node + descendant).

Currently the nodes have a boolean property ( node.left or node.isLeft() ) to indicate if they should grow to the left or to the right. This property changes automatically if they are moved from one side of the root node to the other. All children from a node have the same value than their parent.

Maybe the code can be changed, so the nodes can have a second 'left' property that can be modified in each node indepently, but that only applies if the node is a children from the root node or from a free positioned node. this way if you have a free positioned node that resembles a mini mindmap, when it looses its free positioned property it returns to its parent in a normal way. If you change it again to free positioned, then it remembers which childs should go left and which should go right.

jm2c
edo

**Comment by euu2021** [comment]
Related discussions: [this](https://sourceforge.net/p/freeplane/discussion/758437/thread/6c37341171/)  and [this](https://sourceforge.net/p/freeplane/discussion/758437/thread/284e8eb79d/?limit=25#a60b).

This feature would improve, at least, one of my usecases of Freeplane, which can be described like this: 

I usually use my map growing only to the right side, but, for some kinds of information (eg, debates mapping), I find it very useful to distribute de information on both sides, because it gets easier to visualise and also memorize. Arguments on a debate get distributed on two fields (pro and against), so it gets visually clear when they grow to different sides around a central node.

So, I like to store the information inside my big map, with everything growing to the right side, but I like to visualize some topics distributed to both sides. 

I tried some workflows to make it work (eg, exporting the topic branch, just to visualize it), but it's too cumbersome (eg, it's difficult to edit nodes when I spot mistakes while reading the exported map), so I end up not using it as much as I would like.

**Comment by dpolivaev** [comment]
Implemented, please comment at #482
